vertx-web/src/test/java/io/vertx/ext/web/RouterTest.java

```diff
@@ -97,7 +97,7 @@ private void testRoutePathAndMethod(HttpMethod method, boolean exact) throws Exc
     }
     for (HttpMethod meth: METHODS) {
       if (meth != method) {
-        testRequest(meth, path, 404, "Not Found");
+        testRequest(meth, path, 405, "Method Not Allowed");
```

Comment:
The test is incomplete. According to [rfc7231#section-6.5.5](https://tools.ietf.org/html/rfc7231#section-6.5.5), the system must return the `Allow` header specifying the allowed methods for the resource.
Idem below